frontend_tests/puppeteer_tests/03-compose.js

```diff
@@ -0,0 +1,177 @@
+
+    // The last message(private) in the narrow is selected.
+    // Now we go up and open compose box with r key
+    await page.keyboard.press("Escape");
+    await page.keyboard.press("KeyK");
+    await page.keyboard.press("KeyR");
+    await common.check_form_contents(page, '#send_message_form', {
+        stream_message_recipient_stream: "Verona",
+        stream_message_recipient_topic: "Reply test",
+        content: "",
+    });
+
+    // Test opening and closing of compose box.
+    await page.waitForSelector('#stream-message', {hidden: false});
+    await close_compose_box(page);
+    await page.waitForSelector('#stream-message', {hidden: true});
+
+    await page.keyboard.press('KeyX');
+    await page.waitForSelector('#private-message', {hidden: false});
+    await close_compose_box(page);
+    await page.waitForSelector('#private-message', {hidden: true});
+
+
+    // Test focus after narrowing to PMs with a user and typing 'c'
+    const you_and_cordelia_selector = '*[title="Narrow to your private messages with Cordelia Lear"]';
+    // For some unknown reason page.click() isn't working here.
+    await page.evaluate((selector) => document.querySelector(selector).click(),
+                        you_and_cordelia_selector);
+    const cordelia_user_id = await common.get_user_id_from_name(page, 'Cordelia Lear');
+    const pm_list_selector = `li[data-user-ids-string="${cordelia_user_id}"].expanded_private_message.active-sub-filter`;
+    await page.waitForSelector(pm_list_selector, {visible: true});
+    await close_compose_box(page);
+    await page.keyboard.press('KeyC');
+    await page.waitForSelector('#compose', {visible: true});
+    await page.waitForFunction(() => {
+        return document.activeElement === $('.compose_table #stream_message_recipient_stream')[0];
+    });
+    await close_compose_box(page);
+
+    // Make sure multiple PM recipients display properly.
+    const recipients = ['cordelia@zulip.com', 'othello@zulip.com'];
+    const multiple_recipients_pm = 'A huddle to check spaces';
+    const pm_selector = `.messagebox:contains('${multiple_recipients_pm}')`;
+    await common.send_message(page, 'private', {
+        recipient: recipients.join(', '),
+        outside_view: true,
+        content: multiple_recipients_pm,
+    });
+
+    // Go back to all messages view and make sure all messages are loaded.
+    await page.keyboard.press('Escape');
+
+    await page.waitForSelector('#zhome .message_row', {visible: true});
+    await page.waitForFunction((selector) => {
+        return $(selector).length !== 0;
+    }, {}, pm_selector);
+    await page.evaluate((selector) => {
+        $(selector).slice(-1)[0].click();
+    }, pm_selector);
+    await page.waitForSelector('#compose-textarea', {visible: true});
+    const recipient_internal_emails = [await common.get_internal_email_from_name(page, 'Othello, the Moor of Venice'),
```

Comment:
I actually recommend that `get_internal_email_from_name` has some shortcuts:

* cordelia -> Cordelia Lear
* othello -> Othello, the Moor of Venice
* hamlet -> King Hamlet

Reply:
I wasn't sure if we're going to use it more often. Thought we can add it later if we see ourselves using it more.

Shall I add them now?

Reply:
```
/casper_tests$ git grep -c @zulip.com
02-site.js:4
03-narrow.js:9
04-compose.js:4
05-subscriptions.js:11
08-edit.js:1
13-user-deactivation.js:3
14-drafts.js:3
```
I guess we use them enough number of times for us to have shortcuts(especially subscriptions), I'll add them.
Thanks a lot!

Reply:
Okay, just updated it. Added only those three as they are the most used. We do not use Prospero anywhere and others are just single worded.